opentreemap/treemap/js/src/geocoderInvokeUi.js

```diff
@@ -0,0 +1,37 @@
+"use strict";
+
+var $ = require('jquery'),
+    _ = require('lodash'),
+    geocoder = require('treemap/geocoder'),
+    otmTypeahead = require('treemap/otmTypeahead');
+
+module.exports = function (options) {
+    var $addressInput = $(options.addressInput),
+        geocodeCandidateStream = options.searchTriggerStream
+            .map(getDatum)
+            .filter(_.identity),  // ignore false values
+        gcoder = geocoder(options.config),
+        geocodeResponseStream = gcoder.geocodeStream(geocodeCandidateStream);
+
+    return geocodeResponseStream;
+
+    function getDatum() {
+        var datum = otmTypeahead.getDatum($addressInput),
+            inputText = $addressInput.val();
+        if (datum) {
+            if (datum.text) {
+                if (datum.text.startsWith(inputText)) {
```

Comment:
Perhaps instead of testing for `datum.text` it would be better to test for `datum.magicKey`, which is I think what we really want to branch of off here anyways?